adapters/Vungle/VungleAdapter/GADMAdapterVungleUtils.h

```diff
@@ -13,44 +13,28 @@
-void GADMAdapterVungleMapTableSetObjectForKey(NSMapTable *_Nonnull mapTable,
-                                              id<NSCopying> _Nullable key, id _Nullable value);
-
-/// Safely removes the |object| for |key| in mapTable if |key| is not nil.
-void GADMAdapterVungleMapTableRemoveObjectForKey(NSMapTable *_Nullable mapTable, id _Nullable key);
-
-/// Sets |value| for |key| in |dictionary| if |value| is not nil.
-void GADMAdapterVungleMutableDictionarySetObjectForKey(NSMutableDictionary *_Nonnull dictionary,
-                                                       id<NSCopying> _Nullable key,
-                                                       id _Nullable value);
-
-/// Safely removes the |object| for |key| in userDefaults if |key| is not nil.
-void GADMAdapterVungleUserDefaultsRemoveObjectForKey(NSUserDefaults *_Nonnull userDefaults,
-                                                     id _Nullable key);
-
-/// Removes the object for |key| in |dictionary| if |key| is not nil.
-void GADMAdapterVungleMutableDictionaryRemoveObjectForKey(NSMutableDictionary *_Nonnull dictionary,
-                                                          id<NSCopying> _Nullable key);
-
-/// Return a dictionary of Vungle ad playback options.
-NSDictionary *_Nullable GADMAdapterVunglePlaybackOptionsDictionaryForExtras(
-    VungleAdNetworkExtras *_Nullable vungleAdNetworkExtras);
-
-/// Returns an NSError with code |code| and with NSLocalizedDescriptionKey and
+/// Returns a NSError with code |code| and with NSLocalizedDescriptionKey and
 /// NSLocalizedFailureReasonErrorKey values set to |description|.
 NSError *_Nonnull GADMAdapterVungleErrorWithCodeAndDescription(GADMAdapterVungleErrorCode code,
                                                                NSString *_Nonnull description);
 
-/// Returns a VungleAdSize object from the specified adSize
-VungleAdSize GADMAdapterVungleAdSizeForCGSize(CGSize adSize);
+/// Returns a NSError converted from the error object from Vungle SDK into the AdMob error format
+/// The localized description will contain the vungle error code and the description from the Vungle SDK.
+NSError *_Nonnull GADMAdapterVungleErrorToGADError(GADMAdapterVungleErrorCode code,
+                                                               NSInteger vungleCode,
+                                                               NSString *_Nonnull description);
+
+/// Returns a NSError with code |code| and with NSLocalizedDescriptionKey and
+/// NSLocalizedFailureReasonErrorKey values set to |description|, specifically for invalid placement id.
+NSError *_Nonnull GADMAdapterVungleInvalidPlacementErrorWithCodeAndDescription(void);
+
+/// Returns a NSError with code |code| and with NSLocalizedDescriptionKey and
+/// NSLocalizedFailureReasonErrorKey values set to |description|, specifically for invalid app id.
+NSError *_Nonnull GADMAdapterVungleInvalidAppIdErrorWithCodeAndDescription(void);
```

Comment:
Instead of creating a separate utility method for every adapter error code, we can just create one utility method that takes in an adapter error code.

See https://github.com/googleads/googleads-mobile-ios-mediation/blob/main/adapters/Mintegral/MintegralAdapter/GADMAdapterMintegralUtils.m#L20 for reference.

Reply:
There is a single utility method that takes in an adapter error code.
https://github.com/googleads/googleads-mobile-ios-mediation/blob/f21398afb53030c06ec44785fdb12802b1140d2d/adapters/Vungle/VungleAdapter/GADMAdapterVungleUtils.m#L18

Those utility method are convenience methods because the code and message is the same and it is utilized at multiple locations. These convenience methods use the one utility method. For example.
https://github.com/googleads/googleads-mobile-ios-mediation/blob/f21398afb53030c06ec44785fdb12802b1140d2d/adapters/Vungle/VungleAdapter/GADMAdapterVungleInterstitial.m#L95
https://github.com/googleads/googleads-mobile-ios-mediation/blob/f21398afb53030c06ec44785fdb12802b1140d2d/adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m#L62